Automatically align the erase length to erase size (sector size).
This is a compatibility cvhange to fix issues when upgrading from older
u-boot versions and not resetting the environment.
In this settings the "+" before the length parameter is missing.

diff --git a/drivers/mtd/spi/spi_flash.c b/drivers/mtd/spi/spi_flash.c
--- a/drivers/mtd/spi/spi_flash.c
+++ b/drivers/mtd/spi/spi_flash.c
@@ -202,6 +202,12 @@ int spi_flash_cmd_erase(struct spi_flash
 	u8 cmd[4];
 
 	erase_size = flash->sector_size;
+	if (len % erase_size) {
+		printf("SF: Align erase length (%d) to erase size (%d)",
+			len, erase_size);
+		len = len + erase_size - (len % erase_size);
+		printf(" -> new length %d\n", len);
+	}
 	if (offset % erase_size || len % erase_size) {
 		debug("SF: Erase offset/length not multiple of erase size\n");
 		return -1;
